src/warnet/lnnode.py

```diff
@@ -22,28 +21,28 @@ def status(self) -> RunningStatus:
         return self.warnet.container_interface.get_status(self.tank.index, ServiceType.LIGHTNING)
 
     @exponential_backoff(max_retries=20, max_delay=300)
-    def lncli(self, cmd) -> str:
+    @handle_json
+    def lncli(self, cmd) -> dict:
         cmd = f"lncli --network=regtest {cmd}"
         return self.backend.exec_run(self.tank.index, ServiceType.LIGHTNING, cmd)
 
     def getnewaddress(self):
-        res = json.loads(self.lncli("newaddress p2wkh"))
+        res = self.lncli("newaddress p2wkh")
         return res["address"]
 
     def getURI(self):
-        res = json.loads(self.lncli("getinfo"))
+        res = self.lncli("getinfo")
         return res["uris"][0]
 
     def get_wallet_balance(self):
-        res = json.loads(self.lncli("walletbalance"))
+        res = self.lncli("walletbalance")
         return res
 
+    @handle_json
```

Comment:
why do you need the extra @handle_json here? isn't it already being handled on the `self.lncli()` call?

Reply:
Oh whoops, first i added it to all functions, before extracting it onto `lncli`, nice catch